pytube/contrib/channel.py

```diff
@@ -183,19 +249,119 @@ def _extract_videos(raw_json: str) -> Tuple[List[str], Optional[str]]:
+                                  f"{x['richItemRenderer']['content']['videoRenderer']['videoId']}")
+        except (KeyError, IndexError, TypeError):
+            # Extract id from short videos
+            for x in videos:
+                videos_url.append(f"/watch?v="
+                                  f"{x['richItemRenderer']['content']['reelItemRenderer']['videoId']}")
+
         # remove duplicates
-        return (
-            uniqueify(
-                list(
-                    # only extract the video ids from the video data
-                    map(
-                        lambda x: (
-                            f"/watch?v="
-                            f"{x['gridVideoRenderer']['videoId']}"
-                        ),
-                        videos
-                    )
-                ),
-            ),
-            continuation,
-        )
+        return uniqueify(videos_url), continuation
+
+    @property
+    def views(self) -> int:
+        """Extract view count for channel.
+
+        :return: Channel view count
+        :rtype: int
+        """
+        self.html_url = self.about_url
+        # Get the position of the "about" tab.
+        about_tab_pos = len(self.initial_data['contents']['twoColumnBrowseResultsRenderer']['tabs']) - 2
+        try:
+            views_text = self.initial_data['contents']['twoColumnBrowseResultsRenderer']['tabs'][about_tab_pos][
+                'tabRenderer']['content']['sectionListRenderer']['contents'][0]['itemSectionRenderer'][
+                'contents'][0]['channelAboutFullMetadataRenderer']['viewCountText']['simpleText']
+
+            # "1,234,567 view"
+            count_text = views_text.split(' ')[0]
+            # "1234567"
+            count_text = count_text.replace(',', '')
+            return int(count_text)
+        except KeyError:
+            return 0
+
+    @property
+    @cache
+    def title(self) -> str:
+        """Extract the channel title.
+
+        :return: Channel title (name)
+        :rtype: str
+        """
+        self.html_url = self.channel_url
+        return self.initial_data['metadata']['channelMetadataRenderer']['title']
+
+    @property
+    def description(self) -> str:
+        """Extract the channel description.
+
+        :return: Channel description
+        :rtype: str
+        """
+        self.html_url = self.channel_url
+        return self.initial_data['metadata']['channelMetadataRenderer']['description']
+
+    @property
+    def length(self):
+        """Extracts the approximate amount of videos from the channel.
+
+        :return: Channel videos count
+        :rtype: str
+        """
+        self.html_url = self.channel_url
+        return self.initial_data['header']['c4TabbedHeaderRenderer']['videosCountText']['runs'][0]['text']
+
+    @property
+    @cache
+    def last_updated(self):
+        """Extract the date of the last uploaded video.
+
+        :return: Last video uploaded
```

Comment:
presuming Last video upload date (is it in milliseconds/seconds...?)

Reply:
In this code it returns an approximate date, for example: "3 days ago", I managed to modify the code to extract the exact date of the video, however it cannot identify if the last upload was a short video.

Here is the modified code to extract the exact upload date of the last video.

```python

@property
def last_updated(self) -> Optional[date]:
    """Extract the date of the last uploaded video.

    The API does not provide the date of the last update of the channel,
    so we extract the date of the most recent video published.

    The page's html only provides an approximate date, like "3 months ago",
    so we extract the exact date from the video player.

    :return: Date of last uploaded video.
    :rtype: datetime.date
    """
    self.html_url = self.videos_url

    try:
        # On channels that do not have any video available,
        # it will generate an exception which leads to the return of None
        last_video_content = self.initial_data['contents']['twoColumnBrowseResultsRenderer']['tabs'][1][
            'tabRenderer']['content']['richGridRenderer']['contents'][0]['richItemRenderer']['content'][
            'videoRenderer']
    except (IndexError, KeyError):
        return None
    try:
        # We recovered the player of the last loaded video
        innertube_response = InnerTube(client='WEB').player(last_video_content['videoId'])

        # We extract the exact date of the last uploaded video.
        # Example: 2023-05-11.
        publish_date = innertube_response['microformat']['playerMicroformatRenderer']['publishDate']
        return date.fromisoformat(publish_date)
    except (IndexError, KeyError):
        # If unable to retrieve exact date, return raw string.
        # Example: 1 day ago.
        return last_video_content['publishedTimeText']['simpleText']
```

We can try to extract the data from the last video and the last published short and compare the date of it, but I don't think it would be very interesting.